stock_picking_batch_delivery_carrier/models/stock_picking.py

```diff
@@ -0,0 +1,17 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.constrains('carrier_id')
+    def _check_carrier_id(self):
```

Comment:
@dalonsod Wouldn't using the write method be better than using constraints?

Reply:
if we strictly need to check a value (pass it if it's right, or raise an error otherwise), `api.constrains` is the simplest option. `write` is required e.g. if we also need to update another fields (we shouldn't make create/update operations within a constrains method).

Thus, for this case `constrains` seems to be fine.

Anyway @DantePereyra you could move `for`-`if` statement in a `for`-`filtered` one, and raising an error for all pickings involved together.